en/editor/publish/build-open-data-context.md

```diff
@@ -0,0 +1,113 @@
+- **render/avatar.png**：Header images for display in template project, can be deleted.
+- **engine.js**：source code of Canvas engine
+- **index.js**：Open Data Context project entry file where the Open Data Context is rendered by passing XML text and CSS styles to the Canvas engine
+
+## SubContextView Component Description
+
+The SubContextView component contains two main properties, **Design Resolution** and **FPS**.
+
+![](./build-open-data-context/sub-context-view.png)
+
+The rendering process of the Open Data Context is to render the content onto the sharedCanvas, and then the primary project renders the sharedCanvas as a texture map onto the SubContextView.  
+
+### Design Resolution
+
+The image above sets the component's design resolution to 640 * 960, so at the component onLoad phase, the size of the sharedCanvas will be set to 640 * 960, which means that after the build, the open data project will be rendered on a 640 * 960 off-screen canvas, which requires the maximum width of the element is 640 and the maximum height is 960 when setting style in `style.js`, otherwise the rendered content will exceed the canvas. For example:
+
+```js
+// style.js
+export default {
+    container: {
+        width: 640, // max width
+        height: 960,  // max height
+    },
+}
+```
+
+To avoid this part of the data coupling, we support setting a percentage adaptation to the size, for example:
+
+```js
+// style.js
+export default {
+    container: {
+        width: '100%',
+        height: '100%',
+    },
+}
+```
+
+In the actual rendering process, the sharedCanvas can be interpreted as content and the SubContextView node as a container, and the engine will adopt the adaptation strategy of SHOW ALL to render the sharedCanvas to the screen to avoid the UI distortion caused by stretching during rendering.
+
+![](./build-open-data-context/adaption-1.png)
+
+![](./build-open-data-context/adaption-2.png)
+
+### Setting FPS
+
+This property mainly sets the frequency of updating sharedCanvas to the screen by the primary project, to avoid frequent updating of the Open Data Context texture by the primary project, in order to avoid some performance loss.
+
+## Some recommended practices
+
+1. As mentioned before, the next time you build an Open Data Context project template, it will not be overwritten if the **openDataContext** directory already exists in the build directory, because we don't want to overwrite a project that the developer has customized. Since the build directory is ignored by git by default, if you want to include the **openDataContext** project in your version control, you can consider putting the build project into the build-templates directory. You can refer to [Custom Project Build Process](./custom-project-build-template.md) .
```

Comment:
1. As mentioned before, the next time you build an Open Data Context project template, it will not be overwritten if the **openDataContext** directory already exists in the build directory, because we don't want to overwrite a project that the developer has customized. Since the build directory is ignored by git by default, if you want to include the **openDataContext** project in your version control, you can consider putting the build project into the build-templates directory. You can refer to [Custom Project Build Process](./custom-project-build-template.md) 
-->
1. As previously mentioned, the next time an **Open Data Context Project** template is built, it will not be overwritten if the **openDataContext** directory already exists in the build directory. This prevents overwriting a project that the developer has customized. Since the build directory is ignored by **git** by default, if the **openDataContext** project needs to be included in version control consider putting the built project into the `build-templates` directory. Please refer to the [Custom Project Build Process](./custom-project-build-template.md) documentation.